great_expectations/core/batch.py

```diff
@@ -574,3 +587,180 @@ def head(self, n_rows=5, fetch_all=False):
+    query: Optional[str] = None,
+    path: Optional[str] = None,
+    batch_filter_parameters: Optional[dict] = None,
+    **kwargs,
+) -> Union[BatchRequest, RuntimeBatchRequest]:
+    """Obtain formal BatchRequest typed object from allowed attributes (supplied as arguments).
+    This method applies only to the new (V3) Datasource schema.
+
+    Args:
+        batch_request
+
+        datasource_name
+        data_connector_name
+        data_asset_name
+
+        batch_request
+        batch_data
+        query
+        path
+        runtime_parameters
+        data_connector_query
+        batch_identifiers
+        batch_filter_parameters
+
+        limit
+        index
+        custom_filter_function
+
+        sampling_method
+        sampling_kwargs
+
+        splitter_method
+        splitter_kwargs
+
+        batch_spec_passthrough
+
+        **kwargs
+
+    Returns:
+        (BatchRequest) The formal BatchRequest object
+    """
+
+    if batch_request:
+        if not isinstance(batch_request, BatchRequest):
+            raise TypeError(
+                f"batch_request must be an instance of BatchRequest object, not {type(batch_request)}"
+            )
+        datasource_name = batch_request.datasource_name
+
+    # ensure that the first parameter is datasource_name, which should be a str. This check prevents users
+    # from passing in batch_request as an unnamed parameter.
+    if not isinstance(datasource_name, str):
+        raise ge_exceptions.GreatExpectationsTypeError(
+            f"the first parameter, datasource_name, must be a str, not {type(datasource_name)}"
+        )
+
+    if len([arg for arg in [batch_data, query, path] if arg is not None]) > 1:
+        raise ValueError("Must provide only one of batch_data, query, or path.")
+
+    if any(
+        [
+            batch_data is not None
+            and runtime_parameters
+            and "batch_data" in runtime_parameters,
+            query and runtime_parameters and "query" in runtime_parameters,
+            path and runtime_parameters and "path" in runtime_parameters,
+        ]
+    ):
+        raise ValueError(
+            "If batch_data, query, or path arguments are provided, the same keys cannot appear in the "
+            "runtime_parameters argument."
+        )
+
+    if batch_request:
+        # TODO: Raise a warning if any parameters besides batch_requests are specified
+        return batch_request
+
+    if any([batch_data is not None, query, path, runtime_parameters]):
+        runtime_parameters = runtime_parameters or {}
+        if batch_data is not None:
+            runtime_parameters["batch_data"] = batch_data
+        elif query is not None:
+            runtime_parameters["query"] = query
+        elif path is not None:
+            runtime_parameters["path"] = path
+
+        if batch_identifiers is None:
+            batch_identifiers = kwargs
+        else:
+            # Raise a warning if kwargs exist
```

Comment:
Looks like this is a `TODO`. Is this something that could be easily done in this PR?

Reply:
@NathanFarmer Since this is an earlier comment, I would prefer for it to be handled as part of a dedicated effort.  Thanks!